### PR TITLE
Add a new constructor

### PR DESCRIPTION
**Pull request by colasjojo**
### From NSData to OHPDFDocument.

The new method creates an `OHPDFDocument` from an `NSData`.

I am not sure about the `__bridge` keyword, I have never used it. Actually, I am a newbie for all the `CF` things.

### CODE REVIEW

File: OHPDFImage/OHPDFDocument.m [hunk 1]
```diff
@@ -35,6 +35,21 @@ @implementation OHPDFDocument
 
 #pragma mark - Constructors
 
++ (instancetype)documentWithData:(NSData *)data
+{
+    CGDataProviderRef dataRef = CGDataProviderCreateWithCFData((__bridge CFDataRef)data);
+    CGPDFDocumentRef docRef = CGPDFDocumentCreateWithProvider(dataRef);
```
Comment (AliSoftware) [inline OHPDFImage/OHPDFDocument.m]: Shouldn't you ensure that `data` is not nil (and neither is `dataRef`) here? I'm not sure but this might crash if any of those are `nil` here.

File: OHPDFImage/OHPDFDocument.h [hunk 2]
```diff
@@ -61,6 +61,15 @@
  */
 + (instancetype)documentWithRef:(CGPDFDocumentRef)docRef;
 
+/**
+ *  Create an OHPDFDocument from an NSData created from a PDF file.
+ *
+ *  @param data The NSData reated from the PDF file to load
```
Comment (AliSoftware) [inline OHPDFImage/OHPDFDocument.h]: typo `reated`->`created`

**Comment by colasjojo** [comment]
Like this?

(is it possible to modify the pull request without committing and creating a new PR?)

```
+ (instancetype)documentWithData:(NSData *)data
{
    if (data)
    {
        CGDataProviderRef dataRef = CGDataProviderCreateWithCFData((__bridge CFDataRef)data);
        
        if (dataRef)
        {
            CGPDFDocumentRef docRef = CGPDFDocumentCreateWithProvider(dataRef);
            CGDataProviderRelease(dataRef);
        }
        else
        {
            CGDataProviderRelease(dataRef);
            return nil ;
        }
    }
    else
    {
        return nil ;
    }


    OHPDFDocument* doc = nil;
    if (docRef)
    {
        doc = [self documentWithRef:docRef];
        CGPDFDocumentRelease(docRef);
    }
    return doc;
}
```

**Comment by AliSoftware** [comment]
> (is it possible to modify the pull request without committing and creating a new PR?)
Simply add new commits to your branch, the PR will be updated automatically.

> Like this?

There are some mistakes in your code (did you test it?) as for example if `dataRef` is `nil` then you go into your `else` block which release this `nil` value. This will probably crash if there is no safeguard (I haven't checked the documentation, maybe `CFDataProviderRelease` includes a safeguard itself, but better safe than sorry)


You could also have a more concise code, e.g. like this:

```objc
+ (instancetype)documentWithData:(NSData *)data
{
    if (!data) return nil;
    CGDataProviderRef dataRef = CGDataProviderCreateWithCFData((__bridge CFDataRef)data);

    if (!dataRef) return nil;    
    CGPDFDocumentRef docRef = CGPDFDocumentCreateWithProvider(dataRef);
    CGDataProviderRelease(dataRef);

    if (!docRef) return nil;
    OHPDFDocument* doc = [self documentWithRef:docRef];
    CGPDFDocumentRelease(docRef);

    return doc;
}
```

Another practice is to avoid having multiple calls to `return` and only have one exit point / one return at the end, never in the middle of the method (I let that "as an exercice for the reader" as they always say 😄)

**Comment by colasjojo** [comment]
Thanks ! (no, I haven't tested...)

In this case, I feel that to have `return`s in the middle of the function is more elegant than having multiple `if-then-else`.

**Comment by AliSoftware** [comment]
We should definitely start adding unit tests (#4) at least to test cases when input parameters are `nil` before starting adding more and more features, otherwise we won't guarantee the non-regression of this pod on a long run :-/

**Comment by AliSoftware** [comment]
Version `3.2.1` published with this new constructor + some fix in the `OHVectorImage insets` interpretation when tinting is off.

**Comment by colasjojo** [comment]
Cool ! Honte à moi, je n'ai fait de test unitaire ! J'aimerais vraiment pouvoir t'aider là-dessus. Le mieux pour moi, ça serait que tu push un premier test unitaire et du coup je pourrai m'en inspirer pour en créer d'autres.
